src/main/java/core/basesyntax/util/filereader/FileReaderImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.util.filereader;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+public class FileReaderImpl implements FileReader {
+    @Override
+    public List<String> read(String fromFileName) {
+        File file = new File(fromFileName);
+        Path inputFilePath = Path.of(file.toURI());
+        List<String> lines;
+        try {
```

Comment:
Since Java 7, you can use the `Paths.get()` method directly instead of converting a `File` to a `URI` and then to a `Path`. This would simplify lines 12 and 13 to just one line: `Path inputFilePath = Paths.get(fromFileName);`